### PR TITLE
fix(execution-engine)!: this adds a join behavior for key and value pair used by ap inserting into a map [fixes VM-337]

### PR DESCRIPTION
**Pull request by raftedproc**
fix(execution-engine)!: this adds a join behavior for key and value pair used by ap inserting into a map [fixes VM-337]

### CODE REVIEW

File: air/tests/test_module/instructions/ap.rs [hunk 1]
```diff
@@ -747,3 +747,56 @@ fn ap_canon_stream_map_with_numeric_key_accessor_lambda() {
     ];
     assert_eq!(actual_trace, expected_trace,);
 }
+
+#[test]
+fn ap_map_key_join_behavior() {
+    let vm_1_peer_id = "vm_1_peer_id";
+    let script = format!(
```
Comment (monoid) [inline air/tests/test_module/instructions/ap.rs:754]: Why `format!` here and below?

**Review by monoid — APPROVED** [review]
utACK

File: air/tests/test_module/instructions/ap.rs (hunk 1 above)
Comment (raftedproc) [reply air/tests/test_module/instructions/ap.rs]: Fixed

**Review by mikevoronov — APPROVED** [review]
utACK

**Pull request merged**